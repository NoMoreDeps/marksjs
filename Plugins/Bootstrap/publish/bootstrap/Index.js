"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BsPlugins = void 0;
const Block_Bs_Grid_1 = require("./Block_Bs_Grid");
var Block_Bs_Grid_2 = require("./Block_Bs_Grid");
Object.defineProperty(exports, "BlockBsGridRenderer", { enumerable: true, get: function () { return Block_Bs_Grid_2.BlockBsGridRenderer; } });
const Block_Bs_Button_1 = require("./Block_Bs_Button");
var Block_Bs_Button_2 = require("./Block_Bs_Button");
Object.defineProperty(exports, "BlockBsButtonRenderer", { enumerable: true, get: function () { return Block_Bs_Button_2.BlockBsButtonRenderer; } });
exports.BsPlugins = [
    Block_Bs_Grid_1.BlockBsGridRenderer,
    Block_Bs_Button_1.BlockBsButtonRenderer
];
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiSW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi9zcmMvSW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7O0FBQUEsbURBQTJEO0FBQzNELGlEQUEyRDtBQUFsRCxvSEFBQSxtQkFBbUIsT0FBQTtBQUM1Qix1REFBMkQ7QUFDM0QscURBQTJEO0FBQWxELHdIQUFBLHFCQUFxQixPQUFBO0FBRWpCLFFBQUEsU0FBUyxHQUFHO0lBQ3ZCLG1DQUFtQjtJQUNuQix1Q0FBcUI7Q0FDdEIsQ0FBQyJ9