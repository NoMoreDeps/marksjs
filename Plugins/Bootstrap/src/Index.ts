import { BlockBsGridRenderer }     from "./Block_Bs_Grid"     ;
import { BlockBsButtonRenderer }   from "./Block_Bs_Button"   ;
import { BlockBsCarouselRenderer } from "./Block_Bs_Carousel" ;

export { BlockBsGridRenderer }     from "./Block_Bs_Grid"     ;
export { BlockBsButtonRenderer }   from "./Block_Bs_Button"   ;
export { BlockBsCarouselRenderer } from "./Block_Bs_Carousel" ;

export const BsPlugins = [
  BlockBsGridRenderer   ,
  BlockBsButtonRenderer ,
  BlockBsCarouselRenderer
];
