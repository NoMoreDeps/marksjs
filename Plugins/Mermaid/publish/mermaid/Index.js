"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Block_Mermaid_1 = require("./Block_Mermaid");
Object.defineProperty(exports, "BlockMermaidRenderer", { enumerable: true, get: function () { return Block_Mermaid_1.BlockMermaidRenderer; } });
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiSW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi9zcmMvSW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6Ijs7QUFBQSxpREFBdUQ7QUFBOUMscUhBQUEsb0JBQW9CLE9BQUEifQ==