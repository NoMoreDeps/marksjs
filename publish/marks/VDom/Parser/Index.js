"use strict";
/**
 * Ported to Typescript from original source : https://github.com/creeperyang/html-parser-lite
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.RawHtmlParser = exports.HtmlScanner = exports.HtmlParser = void 0;
const Parser_1 = require("./Parser");
const Scanner_1 = require("./Scanner");
Object.defineProperty(exports, "HtmlScanner", { enumerable: true, get: function () { return Scanner_1.HtmlScanner; } });
class HtmlParser extends Parser_1.HtmlParser {
    constructor(options) {
        options = options || {};
        if (!options.scanner) {
            options.scanner = new Scanner_1.HtmlScanner();
        }
        super(options);
    }
    parse(html) {
        super.parse(html);
        const tree = this.scanner.getRootNode();
        // always reset scanner after parse finish
        this.scanner.reset();
        return tree;
    }
}
exports.HtmlParser = HtmlParser;
const RawHtmlParser = Parser_1.HtmlParser;
exports.RawHtmlParser = RawHtmlParser;
var Node_1 = require("./Node");
Object.defineProperty(exports, "Node", { enumerable: true, get: function () { return Node_1.Node; } });
Object.defineProperty(exports, "NODE_TYPE", { enumerable: true, get: function () { return Node_1.NODE_TYPE; } });
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiSW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi8uLi8uLi9zcmMvVkRvbS9QYXJzZXIvSW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IjtBQUFBOztHQUVHOzs7QUFFSCxxQ0FBa0Q7QUFDbEQsdUNBQWtEO0FBcUI3Qiw0RkFyQloscUJBQVcsT0FxQlk7QUFuQmhDLE1BQU0sVUFBVyxTQUFRLG1CQUFNO0lBQzNCLFlBQVksT0FBYTtRQUNyQixPQUFPLEdBQUcsT0FBTyxJQUFJLEVBQUUsQ0FBQTtRQUN2QixJQUFJLENBQUMsT0FBTyxDQUFDLE9BQU8sRUFBRTtZQUNsQixPQUFPLENBQUMsT0FBTyxHQUFHLElBQUkscUJBQVcsRUFBRSxDQUFBO1NBQ3RDO1FBQ0QsS0FBSyxDQUFDLE9BQU8sQ0FBQyxDQUFBO0lBQ2xCLENBQUM7SUFDRCxLQUFLLENBQUMsSUFBWTtRQUNkLEtBQUssQ0FBQyxLQUFLLENBQUMsSUFBSSxDQUFDLENBQUE7UUFDakIsTUFBTSxJQUFJLEdBQUcsSUFBSSxDQUFDLE9BQVEsQ0FBQyxXQUFXLEVBQUUsQ0FBQTtRQUN4QywwQ0FBMEM7UUFDMUMsSUFBSSxDQUFDLE9BQVEsQ0FBQyxLQUFLLEVBQUUsQ0FBQTtRQUNyQixPQUFPLElBQUksQ0FBQztJQUNoQixDQUFDO0NBQ0o7QUFJUSxnQ0FBVTtBQUZuQixNQUFNLGFBQWEsR0FBRyxtQkFBTSxDQUFDO0FBRUssc0NBQWE7QUFDL0MsK0JBQXlDO0FBQWhDLDRGQUFBLElBQUksT0FBQTtBQUFFLGlHQUFBLFNBQVMsT0FBQSJ9