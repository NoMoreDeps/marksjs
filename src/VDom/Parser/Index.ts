/**
 * Ported to Typescript from original source : https://github.com/creeperyang/html-parser-lite
 */

import { HtmlParser as Parser } from "./Parser"  ;
import { HtmlScanner }          from "./Scanner" ;

class HtmlParser extends Parser {
    constructor(options?: any) {
        options = options || {}
        if (!options.scanner) {
            options.scanner = new HtmlScanner()
        }
        super(options)
    }
    parse(html: string) {
        super.parse(html)
        const tree = this.scanner!.getRootNode()
        // always reset scanner after parse finish
        this.scanner!.reset()
        return tree;
    }
}

const RawHtmlParser = Parser;

export { HtmlParser, HtmlScanner, RawHtmlParser };
export { Node, NODE_TYPE } from "./Node";