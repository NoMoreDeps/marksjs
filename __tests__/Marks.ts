import { MarksRenderer, Plugins } from "../src/Index";

function createRenderer() {
  const r = new MarksRenderer();
  r.registerRenderers(...Plugins.map(_ => new _()));
  return r;
}

function testHtmlAndTextWithSnapshot(r: MarksRenderer, template: string, forceSecondTemplate?: string) {
  let elt = r.render(template);
  expect(elt.outerHTML).toMatchSnapshot(template);
  let eltStr = r.renderToText(template, -1);
  expect(eltStr).toBe(forceSecondTemplate ?? elt.outerHTML);
}

describe("Test dom", () => {
  it("Should be empty", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));

    testHtmlAndTextWithSnapshot(r, ``);
  });
})

describe("Emphasis", () => {
  it("Should render italic", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));

    testHtmlAndTextWithSnapshot(r, `This is *Italic*`);
    testHtmlAndTextWithSnapshot(r, `This is _Italic_`);
  });

  it("Should render bold", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));

    testHtmlAndTextWithSnapshot(r, `This is **Bold**`);
    testHtmlAndTextWithSnapshot(r, `This is __Bold__`);
  });

  it("Should render underline", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));

    testHtmlAndTextWithSnapshot(r, `This is ~underline~`);
  });

  it("Should render stroke", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));

    testHtmlAndTextWithSnapshot(r, `This is ~~stroke~~`);
  });

  it("Should render inline quote", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));

    testHtmlAndTextWithSnapshot(r, `This is \`inline quote\``);
  });

  it("Should render Bold & Italic", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r, `This is __*Bold &amp; Italic*__`);
  });

  it("Should render multiline emphasis", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));

    testHtmlAndTextWithSnapshot(r, `This is *Italic*  
    This is **Bold**  
    This is ~underline~  
    This is ~stroke~  
    This is \`inline quote\``);
  });
});

describe("Ordered lists", () => {
    it("Should render list starting with 1", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));

    testHtmlAndTextWithSnapshot(r,`1. Item 1
    1. Item 2
    1. Item 3`);
  });

  it("Should render list starting with any number from 0 to 9", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r,`0. Item 1
8. Item 2
1. Item 3`);
  });

  it("Should render list starting with the generic symbol", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r,`#. Item 1
#. Item 2
#. Item 3`);
  });


  it("Should render nested list starting with 1", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r,`1. Item 1
1. Item 2
  1. Item 2.1
    1. Item 2.1.1
  1. Item 2.2
1. Item 3`);
  });

  it("Should render nested list starting with any number from 0 to 9", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r,`9. Item 1
1. Item 2
  8. Item 2.1
    2. Item 2.1.1
  6. Item 2.2
0. Item 3`);
  });

  it("Should render nested list starting with the generic symbol", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r,`#. Item 1
#. Item 2
  #. Item 2.1
    #. Item 2.1.1
  #. Item 2.2
#. Item 3`);
  });

  it("Should render nested list starting with the generic symbol without conflicting with Heading", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r,`# Heading
#. Item 1
#. Item 2
  #. Item 2.1
    #. Item 2.1.1
  #. Item 2.2
#. Item 3
# Heading`);
  });
});

describe("Unordered lists", () => {
   it("Should render simple unordered list", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r,`* Item 1
* Item 2
* Item 3`);
  });

  it("Should render nested unordered list", () => {
    const r = new MarksRenderer();
    r.registerRenderers(...Plugins.map(_ => new _()));
    testHtmlAndTextWithSnapshot(r,`* Item 1
* Item 2
  * Item 2.1
    * Item 2.1.1
  * Item 2.2
* Item 3`);
  });
});

describe("Heading", () => {
  it("Should render Heading 1 to 6 with #", () => {
    const r = createRenderer();
    for(let i = 1; i <= 6; i++) {
      testHtmlAndTextWithSnapshot(r,' Heading'.padStart(i + ' Heading'.length, "#"));
    }
  });
  it("Should render Heading 1 to 6 with =", () => {
    const r = createRenderer();
    for(let i = 1; i <= 6; i++) {
      testHtmlAndTextWithSnapshot(r,`Heading
${"".padStart(i , "=")}`);
    }
  });
  it("Should render === block as text if nothing", () => {
    const r = createRenderer();

    testHtmlAndTextWithSnapshot(r,`Text not heading   
    \\=\\=\\=`, "<div><p><span><span>Text not heading   </span><br><span>    &#61;&#61;&#61;</span></span></p></div>");
    
  });
});

describe("Blockquote", () => {
  it("Should render inline quote", () => {
    const r = createRenderer();

    testHtmlAndTextWithSnapshot(r,'> This is a blockquote');
  });

  it("Should render multiline quote", () => {
    const r = createRenderer();

    testHtmlAndTextWithSnapshot(r,`> This is a multiline quotation.  
> Now you know how to do !`);
  });

  it("Should render emphasis inside blockquote", () => {
    const r = createRenderer();

    testHtmlAndTextWithSnapshot(r,`> This text can use **bold** or *italic*  
> and so much ***\`more\`***...`);
  });
});

describe("Task", () => {
  it("Should render todo task", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,'- [ ] Todo task');
  });

  it("Should render done task", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,'- [x] It is done');
  });

  it("Should render tasks", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`- [ ] Todo 1
- [x] Done 1`);
  });
});

describe("Table", () => {
  it("Should render table", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`| Tables        | Are           | Cool  |
|---------------|---------------|-------|
| col 3 is      | right-aligned | $1600 |
| col 2 is      | *centered*    |   $12 |`);
  });

  it("Should render with left / right align", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`|A|B|C|
|--|--:|:--:|
|X|X|X|`);
  });
});

describe("Code block", () => {
  it("Should render code block", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`\`\`\`
  Place some code here
\`\`\``); 
  });

  it("Should render code block with color", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`\`\`\`javascript
  class Hello {
    sayHello() {
      return "Hello Kitty";
    }
  }
\`\`\``);
  });
});

describe("Link / Image / Ruler", () => {
  it("Should render Link", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`[Wikipedia](https://wikipedia.com "new")`);
  });

  it("Should render Image", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`![Wikipedia](https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2.png "Wikipedia Logo")`);
  });

  it("Should render Ruler", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`___`);
  });
});

describe("Image Block", () => {
  it("Should render Image block", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`[img width:100px]{{
      "src"   : "https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2.png",
      "alt"   : "Image",
      "title" : "Me, myself and I"
    }}`);
  });
});


describe("Styles", () => {
  it("Should render theme, variant, classes", () => {
    const r = createRenderer();
    r.setThemeStyle({
      all: {
        theme: {
          dark: ["darkCls"]
        },
        variant: {
          primary: ["primaryCls"]
        }
      }
    })
    testHtmlAndTextWithSnapshot(r,`This is a text ::- classes:myClass1,myClass2 theme:dark variant:primary`);
  });
});

describe("Styles", () => {
  it("Should table block markdown and csv", () => {
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`[table format:markdown] {{
      | COL A | COL B   | COL C  |
      |-------|---------|--------|
      | AA    | BB      |CC      |
    }}`);

    testHtmlAndTextWithSnapshot(r, `[table format:csv] {{
      COL A,COL B,COL C
      AA,BB,CC
      }}`);
  });
});

describe("Marks block", () => {
  it("Should render marks block", () => {
    const expected = "";
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r,`[marks]{{
      ### h3 heading
    }}`);
  });
});

describe("HTML Block", () => {
  it("Should render html bock", () => {
    const expected = "";
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r, `[html]{{
      <h3>H3 heading</h3>
    }}`);
  });
});

describe("Nested elements", () => {
  it("Should render inline nested and block nested", () => {
    const expected = "";
    const r = createRenderer();
    
    testHtmlAndTextWithSnapshot(r, `Hello ::- ref:001

Ceci est un @@001@@`);
  });
});


/**
describe("", () => {
  it("Should ", () => {
    const expected = "";
    const r = createRenderer();
    
    const elt = r.render(``);
    
    expect(elt.outerHTML).toBe(expected);
  });
});
  */
