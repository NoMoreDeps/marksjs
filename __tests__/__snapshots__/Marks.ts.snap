// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote Should render emphasis inside blockquote: > This text can use **bold** or *italic*  
> and so much ***\`more\`***... 1`] = `"<div><p><blockquote><div><p><span><span>This text can use </span><b><span>bold</span></b><span> or </span><em><span>italic</span></em><br><span>and so much </span><b><em><code><span>more</span></code></em></b><span>...</span></span></p></div></blockquote></p></div>"`;

exports[`Blockquote Should render inline quote: > This is a blockquote 1`] = `"<div><p><blockquote><div><p><span><span>This is a blockquote</span></span></p></div></blockquote></p></div>"`;

exports[`Blockquote Should render multiline quote: > This is a multiline quotation.  
> Now you know how to do ! 1`] = `"<div><p><blockquote><div><p><span><span>This is a multiline quotation.  </span><br><span>Now you know how to do !</span></span></p></div></blockquote></p></div>"`;

exports[`Code block Should render code block with color: \`\`\`javascript
  class Hello {
    sayHello() {
      return "Hello Kitty";
    }
  }
\`\`\` 1`] = `
"<div><p><pre><code class=\\"language-javascript\\">  class Hello {
    sayHello() {
      return \\"Hello Kitty\\";
    }
  }</code></pre></p></div>"
`;

exports[`Code block Should render code block: \`\`\`
  Place some code here
\`\`\` 1`] = `"<div><p><pre><code>  Place some code here</code></pre></p></div>"`;

exports[`Emphasis Should render Bold & Italic: This is __*Bold &amp; Italic*__ 1`] = `"<div><p><span><span>This is </span><b><em><span>Bold &amp; Italic</span></em></b></span></p></div>"`;

exports[`Emphasis Should render bold: This is **Bold** 1`] = `"<div><p><span><span>This is </span><b><span>Bold</span></b></span></p></div>"`;

exports[`Emphasis Should render bold: This is __Bold__ 1`] = `"<div><p><span><span>This is </span><b><span>Bold</span></b></span></p></div>"`;

exports[`Emphasis Should render inline quote: This is \`inline quote\` 1`] = `"<div><p><span><span>This is </span><code><span>inline quote</span></code></span></p></div>"`;

exports[`Emphasis Should render italic: This is *Italic* 1`] = `"<div><p><span><span>This is </span><em><span>Italic</span></em></span></p></div>"`;

exports[`Emphasis Should render italic: This is _Italic_ 1`] = `"<div><p><span><span>This is </span><em><span>Italic</span></em></span></p></div>"`;

exports[`Emphasis Should render multiline emphasis: This is *Italic*  
    This is **Bold**  
    This is ~underline~  
    This is ~stroke~  
    This is \`inline quote\` 1`] = `"<div><p><span><span>This is </span><em><span>Italic</span></em><br><span>    This is </span><b><span>Bold</span></b><br><span>    This is </span><span style=\\"text-decoration: underline\\"><span>underline</span></span><br><span>    This is </span><span style=\\"text-decoration: underline\\"><span>stroke</span></span><br><span>    This is </span><code><span>inline quote</span></code></span></p></div>"`;

exports[`Emphasis Should render stroke: This is ~~stroke~~ 1`] = `"<div><p><span><span>This is </span><span style=\\"text-decoration: line-through\\"><span>stroke</span></span></span></p></div>"`;

exports[`Emphasis Should render underline: This is ~underline~ 1`] = `"<div><p><span><span>This is </span><span style=\\"text-decoration: underline\\"><span>underline</span></span></span></p></div>"`;

exports[`HTML Block Should render html bock: [html]{{
      <h3>H3 heading</h3>
    }} 1`] = `"<div><p><div><h3><span>H3 heading</span></h3></div></p></div>"`;

exports[`Heading Should render === block as text if nothing: Text not heading   
    \\=\\=\\= 1`] = `"<div><p><span><span>Text not heading   </span><br><span>    ===</span></span></p></div>"`;

exports[`Heading Should render Heading 1 to 6 with #: # Heading 1`] = `"<div><h1><span> Heading</span></h1></div>"`;

exports[`Heading Should render Heading 1 to 6 with #: ## Heading 1`] = `"<div><h2><span> Heading</span></h2></div>"`;

exports[`Heading Should render Heading 1 to 6 with #: ### Heading 1`] = `"<div><h3><span> Heading</span></h3></div>"`;

exports[`Heading Should render Heading 1 to 6 with #: #### Heading 1`] = `"<div><h4><span> Heading</span></h4></div>"`;

exports[`Heading Should render Heading 1 to 6 with #: ##### Heading 1`] = `"<div><h5><span> Heading</span></h5></div>"`;

exports[`Heading Should render Heading 1 to 6 with #: ###### Heading 1`] = `"<div><h6><span> Heading</span></h6></div>"`;

exports[`Heading Should render Heading 1 to 6 with =: Heading
= 1`] = `"<div><h1><span> Heading</span></h1></div>"`;

exports[`Heading Should render Heading 1 to 6 with =: Heading
== 1`] = `"<div><h2><span> Heading</span></h2></div>"`;

exports[`Heading Should render Heading 1 to 6 with =: Heading
=== 1`] = `"<div><h3><span> Heading</span></h3></div>"`;

exports[`Heading Should render Heading 1 to 6 with =: Heading
==== 1`] = `"<div><h4><span> Heading</span></h4></div>"`;

exports[`Heading Should render Heading 1 to 6 with =: Heading
===== 1`] = `"<div><h5><span> Heading</span></h5></div>"`;

exports[`Heading Should render Heading 1 to 6 with =: Heading
====== 1`] = `"<div><h6><span> Heading</span></h6></div>"`;

exports[`Image Block Should render Image block: [img width:100px]{{
      "src"   : "https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2.png",
      "alt"   : "Image",
      "title" : "Me, myself and I"
    }} 1`] = `"<div><p><img alt=\\"Image\\" title=\\"Me, myself and I\\" src=\\"https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2.png\\"></p></div>"`;

exports[`Link / Image / Ruler Should render Image: ![Wikipedia](https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2.png "Wikipedia Logo") 1`] = `"<div><p><span><img src=\\"https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2.png\\" alt=\\"Wikipedia\\" title=\\"Wikipedia Logo\\"></span></p></div>"`;

exports[`Link / Image / Ruler Should render Link: [Wikipedia](https://wikipedia.com "new") 1`] = `"<div><p><span><a href=\\"https://wikipedia.com\\" target=\\"new\\"><span>Wikipedia</span></a></span></p></div>"`;

exports[`Link / Image / Ruler Should render Ruler: ___ 1`] = `"<div><hr></div>"`;

exports[`Marks block Should render marks block: [marks]{{
      ### h3 heading
    }} 1`] = `"<div><p><div><h3><span> h3 heading</span></h3></div></p></div>"`;

exports[`Nested elements Should render inline nested and block nested: Hello ::- ref:001

Ceci est un @@001@@ 1`] = `"<div><p></p><p><span><span>Ceci est un </span><div id=\\"001\\" data-mk-ref=\\"true\\"><span><span>Hello </span></span></div></span></p></div>"`;

exports[`Ordered lists Should render list starting with 1: 1. Item 1
    1. Item 2
    1. Item 3 1`] = `"<div><p><ol><li><span>Item 1</span><ol><li><span>Item 2</span></li><li><span>Item 3</span></li></ol></li></ol></p></div>"`;

exports[`Ordered lists Should render list starting with any number from 0 to 9: 0. Item 1
8. Item 2
1. Item 3 1`] = `"<div><p><ol><li><span>Item 1</span></li><li><span>Item 2</span></li><li><span>Item 3</span></li></ol></p></div>"`;

exports[`Ordered lists Should render list starting with the generic symbol: #. Item 1
#. Item 2
#. Item 3 1`] = `"<div><p><ol><li><span>Item 1</span></li><li><span>Item 2</span></li><li><span>Item 3</span></li></ol></p></div>"`;

exports[`Ordered lists Should render nested list starting with 1: 1. Item 1
1. Item 2
  1. Item 2.1
    1. Item 2.1.1
  1. Item 2.2
1. Item 3 1`] = `"<div><p><ol><li><span>Item 1</span></li><li><span>Item 2</span><ol><li><span>Item 2.1</span><ol><li><span>Item 2.1.1</span></li></ol></li><li><span>Item 2.2</span></li></ol></li><li><span>Item 3</span></li></ol></p></div>"`;

exports[`Ordered lists Should render nested list starting with any number from 0 to 9: 9. Item 1
1. Item 2
  8. Item 2.1
    2. Item 2.1.1
  6. Item 2.2
0. Item 3 1`] = `"<div><p><ol><li><span>Item 1</span></li><li><span>Item 2</span><ol><li><span>Item 2.1</span><ol><li><span>Item 2.1.1</span></li></ol></li><li><span>Item 2.2</span></li></ol></li><li><span>Item 3</span></li></ol></p></div>"`;

exports[`Ordered lists Should render nested list starting with the generic symbol without conflicting with Heading: # Heading
#. Item 1
#. Item 2
  #. Item 2.1
    #. Item 2.1.1
  #. Item 2.2
#. Item 3
# Heading 1`] = `"<div><h1><span> Heading</span></h1><p><ol><li><span>Item 1</span></li><li><span>Item 2</span><ol><li><span>Item 2.1</span><ol><li><span>Item 2.1.1</span></li></ol></li><li><span>Item 2.2</span></li></ol></li><li><span>Item 3</span></li></ol></p><h1><span> Heading</span></h1></div>"`;

exports[`Ordered lists Should render nested list starting with the generic symbol: #. Item 1
#. Item 2
  #. Item 2.1
    #. Item 2.1.1
  #. Item 2.2
#. Item 3 1`] = `"<div><p><ol><li><span>Item 1</span></li><li><span>Item 2</span><ol><li><span>Item 2.1</span><ol><li><span>Item 2.1.1</span></li></ol></li><li><span>Item 2.2</span></li></ol></li><li><span>Item 3</span></li></ol></p></div>"`;

exports[`Styles Should render theme, variant, classes: This is a text ::- classes:myClass1,myClass2 theme:dark variant:primary 1`] = `"<div><p><span class=\\"darkCls primaryCls myClass1 myClass2\\"><span>This is a text </span></span></p></div>"`;

exports[`Styles Should table block markdown and csv: [table format:csv] {{
      COL A,COL B,COL C
      AA,BB,CC
      }} 1`] = `"<div><p><table><thead><tr><th align=\\"left\\"><span>COL A</span></th><th align=\\"left\\"><span>COL B</span></th><th align=\\"left\\"><span>COL C</span></th></tr></thead><tbody><tr><td align=\\"left\\"><span>AA</span></td><td align=\\"left\\"><span>BB</span></td><td align=\\"left\\"><span>CC</span></td></tr></tbody></table></p></div>"`;

exports[`Styles Should table block markdown and csv: [table format:markdown] {{
      | COL A | COL B   | COL C  |
      |-------|---------|--------|
      | AA    | BB      |CC      |
    }} 1`] = `"<div><p><table><thead><tr><th align=\\"left\\"><span> COL A </span></th><th align=\\"left\\"><span> COL B   </span></th><th align=\\"left\\"><span> COL C  </span></th></tr></thead><tbody><tr><td align=\\"left\\"><span> AA    </span></td><td align=\\"left\\"><span> BB      </span></td><td align=\\"left\\"><span>CC      </span></td></tr></tbody></table></p></div>"`;

exports[`Table Should render table: | Tables        | Are           | Cool  |
|---------------|---------------|-------|
| col 3 is      | right-aligned | $1600 |
| col 2 is      | *centered*    |   $12 | 1`] = `"<div><p><table><thead><tr><th align=\\"left\\"><span> Tables        </span></th><th align=\\"left\\"><span> Are           </span></th><th align=\\"left\\"><span> Cool  </span></th></tr></thead><tbody><tr><td align=\\"left\\"><span> col 3 is      </span></td><td align=\\"left\\"><span> right-aligned </span></td><td align=\\"left\\"><span> $1600 </span></td></tr><tr><td align=\\"left\\"><span> col 2 is      </span></td><td align=\\"left\\"><em><span>centered</span></em></td><td align=\\"left\\"><span>   $12 </span></td></tr></tbody></table></p></div>"`;

exports[`Table Should render with left / right align: |A|B|C|
|--|--:|:--:|
|X|X|X| 1`] = `"<div><p><table><thead><tr><th align=\\"left\\"><span>A</span></th><th align=\\"right\\"><span>B</span></th><th align=\\"center\\"><span>C</span></th></tr></thead><tbody><tr><td align=\\"left\\"><span>X</span></td><td align=\\"right\\"><span>X</span></td><td align=\\"center\\"><span>X</span></td></tr></tbody></table></p></div>"`;

exports[`Task Should render done task: - [x] It is done 1`] = `"<div><p><ul><li style=\\"list-style-type:none\\"><input type=\\"checkbox\\" checked=\\"checked\\" onclick=\\"() => false\\" style=\\"margin-right:5px\\"><span><span>It is done</span></span></li></ul></p></div>"`;

exports[`Task Should render tasks: - [ ] Todo 1
- [x] Done 1 1`] = `"<div><p><ul><li style=\\"list-style-type:none\\"><input type=\\"checkbox\\" onclick=\\"() => false\\" style=\\"margin-right:5px\\"><span><span>Todo 1</span></span></li><li style=\\"list-style-type:none\\"><input type=\\"checkbox\\" checked=\\"checked\\" onclick=\\"() => false\\" style=\\"margin-right:5px\\"><span><span>Done 1</span></span></li></ul></p></div>"`;

exports[`Task Should render todo task: - [ ] Todo task 1`] = `"<div><p><ul><li style=\\"list-style-type:none\\"><input type=\\"checkbox\\" onclick=\\"() => false\\" style=\\"margin-right:5px\\"><span><span>Todo task</span></span></li></ul></p></div>"`;

exports[`Test dom Should be empty 1`] = `"<div><p></p></div>"`;

exports[`Unordered lists Should render nested unordered list: * Item 1
* Item 2
  * Item 2.1
    * Item 2.1.1
  * Item 2.2
* Item 3 1`] = `"<div><p><ul><li><span>Item 1</span></li><li><span>Item 2</span><ul><li><span>Item 2.1</span><ul><li><span>Item 2.1.1</span></li></ul></li><li><span>Item 2.2</span></li></ul></li><li><span>Item 3</span></li></ul></p></div>"`;

exports[`Unordered lists Should render simple unordered list: * Item 1
* Item 2
* Item 3 1`] = `"<div><p><ul><li><span>Item 1</span></li><li><span>Item 2</span></li><li><span>Item 3</span></li></ul></p></div>"`;
